mw-webapp/src/component/table/Table.tsx

```diff
@@ -1,40 +1,58 @@
-import {useEffect, useState} from "react";
+import {useContext} from "react";
 import {
   flexRender,
-  getCoreRowModel,
-  useReactTable,
+  HeaderGroup,
+  RowModel,
 } from "@tanstack/react-table";
-import {columns} from "src/component/table/columns";
-import {getDayReports} from "src/dataAccessLogic/getDayReports";
 import {DayReport} from "src/model/businessModel/DayReport";
 import styles from "src/component/table/Table.module.scss";
 
 /**
- * Table (need update for split component and logic code)
+ * Context's props table users
+ * @public
  */
-export const Table = () => {
-  const [data, setData] = useState<DayReport[]>([]);
+export interface TableUsers {
+  /**
+ * Group headers table
+ */
+  headerGroup: HeaderGroup<DayReport>[];
+  /**
+ * Object rows table
+ */
+  rowModel: RowModel<DayReport>;
+}
 
-  const loadDayReports = async () => {
-    const dayReports = await getDayReports();
-    setData(dayReports);
-  };
+/**
+ * Table's contexts
+ */
+type ContextTable = React.Context<TableUsers | null>
 
-  useEffect(() => {
-    loadDayReports();
-  }, []);
+/**
+ * Table's props
+ */
+interface TableProps {
+  /**
+  * Table's context
+  */
+  context: ContextTable;
```

Comment:
why do we need context as prop?

Reply:
reuseble component, if we should passed another data in table

Reply:
looks strange, maybe we could use just appropriate prop?

Reply:
removed the contextual approach